File: LivePriceBackend/DTOs/User/UserCreateModel.cs
using LivePriceBackend.Constants.Enums;

namespace LivePriceBackend.DTOs.User;

public class UserCreateModel
{
    public required string Name { get; set; }
    public required string Surname { get; set; }
    public required string Username { get; set; }
    public required string Email { get; set; }
    public required string Password { get; set; }
    public required Role Role { get; set; }
    public int? CustomerId { get; set; }
} 

File: live_price_frontend/lib/modules/users/models/user_create_model.dart
import 'user_role.dart';

class UserCreateModel {
  final String username;
  final String password;
  final String email;
  final String name;
  final String surname;
  final UserRole role;
  final int? customerId;

  UserCreateModel({
    required this.username,
    required this.password,
    required this.email,
    required this.name,
    required this.surname,
    required this.role,
    this.customerId,
  });

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'password': password,
      'email': email,
      'name': name,
      'surname': surname,
      'role': role.id,
      'customerId': customerId,
    };
  }
}
